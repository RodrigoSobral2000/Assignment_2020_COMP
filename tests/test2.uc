/**/
/**//**//**//*herois do mar*/
//nobre povo/*nacao valente*///eee imortal...
int main(void){
    int i=1;
    putchar('\n');
    while(i){
        putchar('0');
        putchar('1');
        if(i){
            putchar('\n')
            i=i+0; //COMENTAR10
            i=0*i;          //^
            while(i==0){    //^
               i=i+1; putchar('0'); 
            }
        /*0101010101010101010100*/    
        } else {putchar('1');}
    }
    i=5050-i;
    $$$ //erro
    return 0;
}
/****EOF*****/