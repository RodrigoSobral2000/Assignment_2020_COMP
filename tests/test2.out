INT
ID(main)
LPAR
VOID
RPAR
LBRACE
INT
ID(i)
ASSIGN
INTLIT(1)
SEMI
ID(putchar)
LPAR
CHRLIT('\n')
RPAR
SEMI
WHILE
LPAR
ID(i)
RPAR
LBRACE
ID(putchar)
LPAR
CHRLIT('0')
RPAR
SEMI
ID(putchar)
LPAR
CHRLIT('1')
RPAR
SEMI
IF
LPAR
ID(i)
RPAR
LBRACE
ID(putchar)
LPAR
CHRLIT('\n')
RPAR
ID(i)
ASSIGN
ID(i)
INTLIT(+0)
SEMI
ID(i)
ASSIGN
INTLIT(0)
MUL
ID(i)
SEMI
WHILE
LPAR
ID(i)
EQ
INTLIT(0)
RPAR
LBRACE
ID(i)
ASSIGN
ID(i)
INTLIT(+1)
SEMI
ID(putchar)
LPAR
CHRLIT('0')
RPAR
SEMI
RBRACE
RBRACE
ELSE
LBRACE
ID(putchar)
LPAR
CHRLIT('1')
RPAR
SEMI
RBRACE
RBRACE
ID(i)
ASSIGN
INTLIT(5050)
MINUS
ID(i)
SEMI
Line 21, col 5: illegal character ($)
Line 21, col 6: illegal character ($)
Line 21, col 7: illegal character ($)
RETURN
INTLIT(0)
SEMI
RBRACE